Dynatrace/dynatrace-operator
# Refactor component feature and version labels

# Description
The components (activegate and oneagent) managed by the operator should have their version included in the labels.

Following labels/annotations were changed:
- Set `app.kubernetes.io/name` to `oneagent` or `activegate`
- Set `app.kubernetes.io/version` to oneagent or activegate version
- Replaced `component.dynatrace.com/feature` with `app.kubernetes.io/component` for the oneagent mode or activegate capability
- Removed version annotation on activegate/oneagent pods

## How can this be tested?
Deploy a dynakube with any oneagent mode and any activegate capability enabled.

Expected labels after this change:
- oneagent:
```yaml
  labels:
    app.kubernetes.io/name: oneagent
    app.kubernetes.io/component: cloudnativefullstack
    app.kubernetes.io/version: 1.241.0.20220413-171953
    app.kubernetes.io/created-by: dynakube
    app.kubernetes.io/managed-by: dynatrace-operator
    
```

```yaml
  selector:
    matchLabels:
      app.kubernetes.io/name: oneagent
      app.kubernetes.io/created-by: dynakube
      app.kubernetes.io/managedy-by: dynatrace-operator
      
```
- activegate:
```yaml
  labels:
    app.kubernetes.io/name: activegate
    app.kubernetes.io/component: activegate
    app.kubernetes.io/version: 1.241.0.20220414-004007
    app.kubernetes.io/created-by: dynakube
    app.kubernetes.io/managed-by: dynatrace-operator
    
```

```yaml
  selector:
    matchLabels:
      app.kubernetes.io/name: activegate
      app.kubernetes.io/created-by: dynakube
      app.kubernetes.io/managedy-by: dynatrace-operator
      
```
- other resources (e.g. secret) that are connected to oneagent/activegate but shouldn't change between versions
```yaml
labels:
    app.kubernetes.io/component: activegate
    app.kubernetes.io/created-by: dynakube
    app.kubernetes.io/name: dynatrace-operator
    
```

## Checklist
